cypress/integration/Pagination.spec.ts

```diff
@@ -0,0 +1,239 @@
+            cy.findByLabelText(/Page 6$/).should('exist');
+            cy.findByLabelText(/Page 10$/).should('exist');
+          });
+        });
+
+        context('when page 6 is selected', () => {
+          beforeEach(() => {
+            cy.findByLabelText('Page 10').click();
+            cy.findByLabelText('Page 6').click();
+          });
+
+          it('should show pages 4 through 8 and page 10', () => {
+            cy.findByLabelText(/Page 4$/).should('exist');
+            cy.findByLabelText(/Page 5$/).should('exist');
+            cy.findByLabelText(/Page 6$/).should('exist');
+            cy.findByLabelText(/Page 7$/).should('exist');
+            cy.findByLabelText(/Page 8$/).should('exist');
+            cy.findByLabelText(/Page 10$/).should('exist');
+          });
+        });
+
+        context('when page 7 is selected', () => {
+          beforeEach(() => {
+            cy.findByLabelText('Page 10').click();
+            cy.findByLabelText('Page 7').click();
+          });
+
+          it('should show page 1 and pages 6 through 10', () => {
+            cy.findByLabelText(/Page 1$/).should('exist');
+            cy.findByLabelText(/Page 6$/).should('exist');
+            cy.findByLabelText(/Page 7$/).should('exist');
+            cy.findByLabelText(/Page 8$/).should('exist');
+            cy.findByLabelText(/Page 9$/).should('exist');
+            cy.findByLabelText(/Page 10$/).should('exist');
+          });
+        });
+      });
+    });
+  });
+
+  context('given Default pagination story is rendered', () => {
+    context('when screen width is larger than 500', () => {
+      beforeEach(() => {
+        h.stories.load('Labs/Pagination', 'With Go To');
+      });
+
+      it('should not have any axe errors', () => {
+        cy.checkA11y();
+      });
+
+      context('when 50 is entered into the Go To input', () => {
+        beforeEach(() => {
+          cy.findByLabelText('Go To')
+            .type('50')
+            .type('{enter}');
+        });
+
+        it('should set page 50 as the selected page', () => {
+          cy.findByText('50').should('have.attr', 'aria-label', 'Selected, Page 50');
+        });
+      });
+
+      context('when -1 is entered into the Go To input', () => {
+        beforeEach(() => {
+          cy.findByLabelText('Go To')
+            .type('-1')
+            .type('{enter}');
+        });
+
+        it('should not change the selected page', () => {
+          cy.findByText('1').should('have.attr', 'aria-label', 'Selected, Page 1');
+        });
+      });
+    });
+  });
+
+  context('when screen width is smaller than 500', () => {
+    beforeEach(() => {
+      cy.viewport(414, 736); // iPhone 6/7/8 Plus
+      h.stories.load('Labs/Pagination', 'Default');
+    });
+
+    it('should show 2 page buttons', () => {
+      cy.findAllByLabelText(/Page \d+$/).should('have.length', 2);
+    });
+
+    context('when the last page is clicked', () => {
```

Comment:
I don't think we've tested `should disable next page button` yet. This spot may be a good place to add it.